core/client/fs/src/main/java/alluxio/client/block/stream/LocalFilePacketWriter.java

```diff
@@ -46,37 +45,62 @@
   private static final long WRITE_TIMEOUT_MS =
       Configuration.getMs(PropertyKey.USER_NETWORK_NETTY_TIMEOUT_MS);
 
-  private final FileSystemContext mContext;
-  private final WorkerNetAddress mAddress;
   private final Channel mChannel;
   private final LocalFileBlockWriter mWriter;
   private final long mBlockId;
   private final long mPacketSize;
   private final ProtoMessage mCreateRequest;
   private final NettyRPCContext mNettyRPCContext;
   private final OutStreamOptions mOptions;
-  private final Closer mCloser = Closer.create();
+  private final Closer mCloser;
 
   /** The position to write the next byte at. */
   private long mPos;
   /** The number of bytes reserved on the block worker to hold the block. */
   private long mPosReserved;
 
   private boolean mClosed = false;
+
   /**
    * Creates an instance of {@link LocalFilePacketWriter}. This requires the block to be locked
    * beforehand.
    *
    * @param context the file system context
    * @param address the worker network address
    * @param blockId the block ID
-   * @param packetSize the packet size
    * @param options the output stream options
    * @return the {@link LocalFilePacketWriter} created
    */
-  public static LocalFilePacketWriter create(FileSystemContext context, WorkerNetAddress address,
-      long blockId, long packetSize, OutStreamOptions options) throws IOException {
-    return new LocalFilePacketWriter(context, address, blockId, packetSize, options);
+  public static LocalFilePacketWriter create(final FileSystemContext context,
+      final WorkerNetAddress address,
+      long blockId, OutStreamOptions options) throws IOException {
+    long packetSize = Configuration.getBytes(PropertyKey.USER_LOCAL_WRITER_PACKET_SIZE_BYTES);
+
+    Closer closer = Closer.create();
+    try {
```

Comment:
move the exception throwing logic up to the factory and leave the constr exception-free